controllers/machine_controller_noderef_test.go

```diff
@@ -19,106 +19,128 @@ package controllers
 import (
 	"testing"
 
-	. "github.com/onsi/gomega"
+	"sigs.k8s.io/controller-runtime/pkg/handler"
+	"sigs.k8s.io/controller-runtime/pkg/reconcile"
+
+	ctrl "sigs.k8s.io/controller-runtime"
 
+	. "github.com/onsi/gomega"
 	corev1 "k8s.io/api/core/v1"
 	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
-	"k8s.io/client-go/kubernetes/scheme"
-	"k8s.io/client-go/tools/record"
 	clusterv1 "sigs.k8s.io/cluster-api/api/v1alpha4"
 	"sigs.k8s.io/cluster-api/controllers/noderefutil"
+	"sigs.k8s.io/cluster-api/controllers/remote"
+	"sigs.k8s.io/cluster-api/util"
 	"sigs.k8s.io/controller-runtime/pkg/client"
-	"sigs.k8s.io/controller-runtime/pkg/client/fake"
+	"sigs.k8s.io/controller-runtime/pkg/log"
```

Comment:
Do we need to use the global logger here?

Reply:
It's being used to pass a NullLogger here https://github.com/kubernetes-sigs/cluster-api/pull/4521/files#diff-7b35b89626a4594a7d7296ab37621dd6998ee1068830357e042c12f00967d0c1R110

Happy to change if there's a more reasonable way.